### PR TITLE
[Core] Convert warning to error in object factory

### PR DESCRIPTION
**Pull request by alxbilger**
I converted the warning to an error in the following code of the ObjectFactor:
```cpp
        if (isUserTemplateNameInTemplateList)
        {
            msg_warning(object.get()) << "Requested template '" << usertemplatename << "' "
                                      << "is not compatible with the current context. "
                                      << "Falling back to the first compatible template: '"
                                      << object->getTemplateName() << "'.";
        }
        else
        {
            msg_warning(object.get()) << "Requested template '" << usertemplatename << "' "
                                      << "cannot be found in the list of available templates [" << ss.str() << "]. "
                                      << "Falling back to default template: '"
                                      << object->getTemplateName() << "'.";
        }
```

To me, this situation is to be taken into account more seriously than a warning.

As failing template deduction is now an error, it leads to a lot of failing tests. I fixed them as well.

I added unit tests for `OglModel`. I factorize the parse code from the 3 masses.

______________________________________________________

By submitting this pull request, I acknowledge that  
**I have read, understand, and agree [SOFA Developer Certificate of Origin (DCO)](https://github.com/sofa-framework/sofa/blob/master/CONTRIBUTING.md#sofa-developer-certificate-of-origin-dco)**.
______________________________________________________

**Reviewers will merge this pull-request only if**  
- it builds with SUCCESS for all platforms on the CI.
- it does not generate new warnings.
- it does not generate new unit test failures.
- it does not generate new scene test failures.
- it does not break API compatibility.
- it is more than 1 week old (or has fast-merge label).

### CODE REVIEW

**Comment by damienmarchal** [comment]
Looks ok to me.

EDIT: @alxbilger want to remove the new errors.

**Comment by alxbilger** [comment]
[ci-build][with-scene-tests]

**Comment by alxbilger** [comment]
@fredroy Was there a mechanism to support `DiagonalMass template="Vec3d"`? Or does it only fall back to `DiagonalMass template="Vec3d,Vec3d"` because it is the default template?

File: Sofa/framework/Core/src/sofa/core/ObjectFactory.cpp [hunk 1]
```diff
@@ -288,14 +288,14 @@ objectmodel::BaseObject::SPtr ObjectFactory::createObject(objectmodel::BaseConte
...
                                       << "is not compatible with the current context. "
                                       << "Falling back to the first compatible template: '"
                                       << object->getTemplateName() << "'.";
         }
         else
         {
-            msg_warning(object.get()) << "Requested template '" << usertemplatename << "' "
+            msg_error(object.get()) << "Requested template '" << usertemplatename << "' "
                                       << "cannot be found in the list of available templates [" << ss.str() << "]. "
                                       << "Falling back to default template: '"
```
Comment (alxbilger) [inline Sofa/framework/Core/src/sofa/core/ObjectFactory.cpp]: I am not sure it's necessarily the default template. To be checked

**Comment by fredroy** [comment]
> @fredroy Was there a mechanism to support `DiagonalMass template="Vec3d"`? Or does it only fall back to `DiagonalMass template="Vec3d,Vec3d"` because it is the default template?

Nope, there is a check to see if the 2nd template is the previous MassType (removed) but there is no mechanism to implicitly suppose the geometry type to be like the data type of the mass (which is like 99.999% of the time actually) 
Do you think it would be a good idea to do it ? (aka if no 2nd template, then implictely suppose it is the same as the the 1st one)

**Comment by alxbilger** [comment]
@fredroy I prefer not adding another implicit complex mechanism for templates. If there are two templates, the user must set 2 templates. If the user set only one, there is an error message containing the list of all available templates. It should give a hint about the fact that there are two templates to provide.

**Comment by alxbilger** [comment]
[ci-build][force-full-build][with-all-tests]

**Comment by alxbilger** [comment]
Some `image` plugin scenes fail because the template used is not compiled in the standard set. It needs the full set. Only the standard set is compiled on the CI.

Decision to take:
1) The failing scenes are ignored on the CI
2) The failing scenes are modified to use a templated available in the standard set
3) The CI compiled the full set

I am in favor of the third option, but we already talked about it in https://github.com/sofa-framework/ci/pull/15. It seems to take too much time at the compilation.

**Comment by fredroy** [comment]
> Some `image` plugin scenes fail because the template used is not compiled in the standard set. It needs the full set. Only the standard set is compiled on the CI.
> 
> Decision to take:
> 
>     1. The failing scenes are ignored on the CI
> 
>     2. The failing scenes are modified to use a templated available in the standard set
> 
>     3. The CI compiled the full set
> 
> 
> I am in favor of the third option, but we already talked about it in [sofa-framework/ci#15](https://github.com/sofa-framework/ci/pull/15). It seems to take too much time at the compilation.

I think there is a 4th solution, adding the ImageB type in the "standard" set, as it is seems quite important if lots of scenes need it after all

**Comment by alxbilger** [comment]
@fredroy Excellent suggestion. I'll do that. Thanks

**Comment by alxbilger** [comment]
@fredroy I applied your suggestion, but there are remining failing scenes. I suggest to ignore them on the CI.

**Review by fredroy — APPROVED** [review]
Looks good to me, (even if I had a quick look on the 100+ scenes 😅)
Should we flag it as breaking (in the sense it may "break" tests somewhere), and/or  "highlighted in the next release"?

**Comment by alxbilger** [comment]
@hugtalbot Can you remind me why you switch this pull request back to wip?

**Comment by fredroy** [comment]
If I remember, just because there are conflicts to solve

**Comment by alxbilger** [comment]
[ci-build][with-scene-tests]

**Comment by alxbilger** [comment]
[ci-build][force-full-build][with-all-tests]

**Comment by fredroy** [comment]
[ci-build][force-full-build][with-all-tests]

**Comment by fredroy** [comment]
Conflicts were solved, tests are OK, and it has been long enough this PR is opened;ready to merge

**Pull request merged**